DNN Platform/Modules/CoreMessaging/Services/MessagingServiceController.cs

```diff
@@ -1,440 +1,556 @@
-
-        public class ReplyDTO : ConversationDTO
-        {
-            public string Body { get; set; }
-
-            public IList<int> FileIds { get; set; }
-        }
-    }
-}
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information
+
+namespace DotNetNuke.Modules.CoreMessaging.Services
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Dynamic;
+    using System.Net;
+    using System.Net.Http;
+    using System.Web;
+    using System.Web.Http;
+
+    using DotNetNuke.Common;
+    using DotNetNuke.Common.Utilities;
+    using DotNetNuke.Entities.Modules;
+    using DotNetNuke.Entities.Portals;
+    using DotNetNuke.Entities.Users;
+    using DotNetNuke.Instrumentation;
+    using DotNetNuke.Modules.CoreMessaging.ViewModels;
+    using DotNetNuke.Security;
+    using DotNetNuke.Services.Localization;
+    using DotNetNuke.Services.Social.Messaging;
+    using DotNetNuke.Services.Social.Messaging.Internal;
+    using DotNetNuke.Services.Social.Notifications;
+    using DotNetNuke.Web.Api;
+
+    /// <summary>
+    /// Provides messaging web services.
+    /// </summary>
+    [SupportedModules("DotNetNuke.Modules.CoreMessaging")]
+    [DnnModuleAuthorize(AccessLevel = SecurityAccessLevel.View)]
+    [DnnAuthorize]
+    public class MessagingServiceController : DnnApiController
+    {
+        private static readonly ILog Logger = LoggerSource.Instance.GetLogger(typeof(MessagingServiceController));
+
+        /// <summary>
+        /// Provides access to the user inbox.
+        /// </summary>
+        /// <param name="afterMessageId">After which message id to start returning new messages.</param>
+        /// <param name="numberOfRecords">How many messges to get.</param>
+        /// <returns>A <see cref="DotNetNuke.Services.Social.Messaging.Internal.Views.MessageBoxView"/>.</returns>
+        [HttpGet]
+        public HttpResponseMessage Inbox(int afterMessageId, int numberOfRecords)
+        {
+            try
+            {
+                var messageBoxView = InternalMessagingController.Instance.GetRecentInbox(this.UserInfo.UserID, afterMessageId, numberOfRecords);
+                var portalId = PortalController.GetEffectivePortalId(UserController.Instance.GetCurrentUserInfo().PortalID);
+
+                messageBoxView.TotalNewThreads = InternalMessagingController.Instance.CountUnreadMessages(this.UserInfo.UserID, portalId);
+                messageBoxView.TotalConversations = InternalMessagingController.Instance.CountConversations(this.UserInfo.UserID, portalId);
+
+                return this.Request.CreateResponse(HttpStatusCode.OK, messageBoxView);
+            }
+            catch (Exception ex)
+            {
+                var message = "An unexpected error occurred while trying to fetch the inbox, consult the server logs for more information.";

```

Comment:
Does this message display to end users?  It'll need to be loclizable, if so.

Reply:
Well IMO, it was not localized to begin with. My goal here was for security reasons to not display the whole details of a general exception which could potentially contain sensitive information. So going from an exception dump to a non-localized human-readable and safer string IMO is a step forward. It would obviously be nice to localize them but I think it may be a separate issue beyound the scope of reducing the build warnings.

Reply:
Are we ok with this, or do we need to add localized exception messages everywhere ? If so I would prefer we make that a separate issue or TODOs for a second run, the goal of this PR (an the others) was really to reduce the build warnings, I just rewrote this because I do believe it is bad practice to send details of general exceptions to the frontend...

Reply:
@valadas we should probably create an issue to localize these exception messages, but let this PR pass for the intended purpose.